src/app/client/src/app/modules/dashboard/components/course-dashboard/course-dashboard.component.ts

```diff
@@ -1,28 +1,120 @@
+import { NavigationHelperService, ResourceService, ToasterService } from '@sunbird/shared';
+import { IImpressionEventInput } from '@sunbird/telemetry';
+import { UserService } from '@sunbird/core';
 import { Subject } from 'rxjs';
-import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Component, OnInit, OnDestroy, AfterViewInit } from '@angular/core';
 import { IDashboard } from '../../interfaces';
 import * as _ from 'lodash-es';
 import { CourseProgressService } from '../../services';
+import { ActivatedRoute, Router } from '@angular/router';
+import {  } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+
 @Component({
   selector: 'app-course-dashboard',
   templateUrl: './course-dashboard.component.html',
   styleUrls: ['./course-dashboard.component.scss']
 })
 export class CourseDashboardComponent implements OnInit, OnDestroy {
-  batchList = [];
-  dashBoardData: IDashboard;
+
+  public dashBoardItems: IDashboard;
+
   public unsubscribe$ = new Subject<void>();
+  courseId: string;
+  queryParams: {};
+  telemetryImpression: IImpressionEventInput;
+  state: {};
 
-  constructor(private courseProgressService: CourseProgressService) { }
+  constructor(
+    private courseProgressService: CourseProgressService,
+    private activatedRoute: ActivatedRoute,
+    private userService: UserService,
+    private router: Router,
+    private navigationhelperService: NavigationHelperService,
+    public resourceService: ResourceService,
+    private toasterService: ToasterService,
+    ) { }
 
   ngOnInit() {
-  this.getBatchList();
+    this.state = history.state;
+    this.initializeFields();
+    this.courseId = _.get(this.activatedRoute.snapshot.params, 'courseId') || _.get(this.state, 'id');
+    this.queryParams = this.activatedRoute.snapshot.queryParams;
+    this.getBatchList();
+    this.setImpressionEvent();
+  }
+
+  // To initialize the dashboard data
+  initializeFields() {
+    this.resourceService.languageSelected$.pipe(takeUntil(this.unsubscribe$))
+    .subscribe(item => {
+    this.dashBoardItems = {
+      totalBatches: {
```

Comment:
This should be IDashboard instance. The title is already part of each object, then why do you want 'totalBatches' as another key in dashboardItems?
Yout HTML should be for loop of this.dashBoardItems & show the cards in the same object with type of card.

Reply:
This is to assign the string data